### ISSUE TITLE
400 Bad Request on AWS Elasticsearch datasource

### ISSUE THREAD

**Issue by tinaboyce** [grafana/grafana#16321]
**What happened**: 
- Getting a 400 Bad Request when trying to use AWS Elasticsearch data source when setting up a query for the graph. Query Inspector also gives 400 error.
- Had Green status when setting up the data source.

**What you expected to happen**:
- No issue setting up the dashboard

**How to reproduce it (as minimally and precisely as possible)**:
- Datasource is configured with Green status but with suspicious console logging. It is AWS Elasticsearch instance with VPC access, no auth required, with dockerized grafana hosted in the same VPC.
![Screen Shot 2019-04-01 at 4 33 59 pm](https://user-images.githubusercontent.com/12929562/55305421-14140b00-549c-11e9-9f32-bd85f0f30a7e.png)
- Trying to select Datasource results in "400 Bad Request", same with selecting Query Inspector.
![Screen Shot 2019-04-01 at 4 38 01 pm](https://user-images.githubusercontent.com/12929562/55305521-98ff2480-549c-11e9-95c4-029fc3c1ece1.png)
- I also have AWS CloudWatch which works without an issue, no 400 here.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v6.0.2
- Data source type & version: AWS Elasticsearch v6.4, VPC access, no auth
- OS Grafana is installed on: Dockerized
- User OS & Browser: Mac OS + Chrome

**Comment by tinaboyce** [comment]
Sorry, forgot to add, Access is "Server (Default)" and all checkbox in Auth is unchecked.

**Comment by marefr** [comment]
@tinaboyce Check the response tab for details about 400 error and report back here. Thanks

**Comment by tinaboyce** [comment]
Thanks @marefr, here's the response. 
<img width="709" alt="Screen Shot 2019-04-01 at 6 56 07 pm" src="https://user-images.githubusercontent.com/12929562/55311910-f6e93780-54af-11e9-8710-da44afdac444.png">

Not sure if it helps but if I just pasted the URL in the browser (which I know is a GET, not a POST), I get a JSON response 
`{"error":{"root_cause":[{"type":"parse_exception","reason":"request body or source parameter is required"}],"type":"parse_exception","reason":"request body or source parameter is required"},"status":400}`

**Comment by marefr** [comment]
The nginx reverse proxy you have in between - is that something you've set up or provided by AWS elasticsearch service? Can you please include the request body/payload sent from Grafana. Thanks

**Comment by tinaboyce** [comment]
Yes, I do have a nginx in front (as a WAF). Usually it doesn't reach there unless the endpoint/path does not exists.

The request body is 
```
{"search_type":"query_then_fetch","ignore_unavailable":true,"index":"*removed*","max_concurrent_shard_requests":256}
{"size":0,"query":{"bool":{"filter":[{"range":{"@timestamp":{"gte":"1554037200000","lte":"1554645599999","format":"epoch_millis"}}},{"query_string":{"analyze_wildcard":true,"query":"*"}}]}},"aggs":{"2":{"date_histogram":{"interval":"15m","field":"@timestamp","min_doc_count":0,"extended_bounds":{"min":"1554037200000","max":"1554645599999"},"format":"epoch_millis"},"aggs":{}}}}
```

**Comment by marefr** [comment]
Cannot reproduce "suspicious console logging" in Grafana v6.1-pre so that should be fixed by upgrading to v6.1 stable when that is released.

Tried your query directly (using POST to _msearch) against a Elasticsearch 6.2.4 instance and it works as expected:
![image](https://user-images.githubusercontent.com/1668778/55383807-56cff480-5529-11e9-9e16-79e37972b2ca.png)

Have no Elasticsearch 6.4 instance available currently. Are you sure you did a POST request to _msearch - please try that. I'm using postman in screenshot, but you can use curl as well:

```bash
curl -H "Content-Type: application/json" -X POST http://localhost:11200/_msearch -d '{"search_type":"query_then_fetch","ignore_unavailable":true,"index":"metrics-2019.04.02","max_concurren
t_shard_requests":256}{"size":0,"query":{"bool":{"filter":[{"range":{"@timestamp":{"gte":"1554037200000","lte":"1554645599999","format":"epoch_millis"}}},{"query_string":{"analyze_wildcard":
true,"query":"*"}}]}},"aggs":{"2":{"date_histogram":{"interval":"15m","field":"@timestamp","min_doc_count":0,"extended_bounds":{"min":"1554037200000","max":"1554645599999"},"format":"epoch_m
illis"},"aggs":{}}}}
'
```

**Comment by torkelo** [comment]
Closing this due to lack of new info / activity
